loadbalancer/zk/publisher_test.go

```diff
@@ -141,6 +151,7 @@ func TestBadPublisherCreate(t *testing.T) {
 }
 
 type fakeClient struct {
+	sync.Mutex
```

Comment:
Do you intend the mutex to be lock/unlockable from outside of member methods? If not, please name it with an unexported name, i.e.

```
type fakeClient struct {
   mtx sync.Mutex
   // ...
}
```